analysis/02_new_model_input/02.2_assign_builing_damage_togrid.md

```diff
@@ -0,0 +1,147 @@
+# Assignment of building damage data to grids from municipalities
+
+```python
+%load_ext jupyter_black
+import pandas as pd
+import os
+from pathlib import Path
+pd.set_option("display.float_format", lambda x: "%.5f" % x)
+input_dir = (
+    Path(os.getenv("STORM_DATA_DIR")) / "analysis/02_new_model_input/02_housing_damage/input/"
+)
+baseline_input_dir = (
+    Path(os.getenv("STORM_DATA_DIR")) / "analysis/01_baseline_model/input/"
+)
+output_dir = (
+    Path(os.getenv("STORM_DATA_DIR")) / "analysis/02_new_model_input/02_housing_damage/output/"
+)
+```
+
+```python
+# reading in file with building damage data and adm3 in grid
+adm3_perc_ingrid = pd.read_csv(
+    input_dir / "Adm3_Perc_inGrid_Weight_Matrix.csv"
+)
+perc_build_dmg_data = pd.read_csv(
+    baseline_input_dir / "combined_input_data.csv"
+)
+build_dmg_data = pd.read_csv(input_dir / "IMpact_data_philipines_SEP_2021.csv")
+phl_north_build = pd.read_csv(
+    input_dir / "phl_north_hotosm_bld_centroid_count.csv"
+)
+phl_south_build = pd.read_csv(
+    input_dir / "phl_south_hotosm_bld_centroid_count.csv"
+)
+adm3_perc_ingrid.shape
+```
+
+```python
+## Section describing the merging of the north and south buildings from HOTOSM shapefile
+# merging the north and south values as the shapefiles are separate
+merged_df_points = (
+    phl_north_build[["id", "NUMPOINTS"]].set_index(["id"])
+    + phl_south_build[["id", "NUMPOINTS"]].set_index(["id"])
+).reset_index()
+merged_df_points = pd.merge(
+    merged_df_points, phl_north_build.drop("NUMPOINTS", axis=1), on="id"
+)
+merged_df_points
+```
+
+```python
+merged_df_points.to_csv(
+    input_dir / "phl_hotosm_bld_centroid_count.csv", index=False
+)
+```
+
+```python
+### Section describing the computation of the building damage percentage
+# combining building damage data and grid information
+merged_total_damage_df = adm3_perc_ingrid.merge(
+    build_dmg_data,
+    left_on="ADM3_PCODE",
+    right_on="pcode",
+    how="left",
+)
+merged_total_damage_df["Totally"].describe()
```

Comment:
When I look at `adm3_perc_ingrid` , it seems that the percentages for a single municipality don't add up to 100% -- if I understand correctly, the column "Grid Completeness" should add up to 100% for a single grid cell, and tells how much each municipality contributes. This will be needed to translate grid based data back to municipality based, however, for municipality -> grid, I think we need to know for each municipality, what percentage is in each grid cell, which should add up to 100% per municipality. I hope that makes sense. 

You should check that `build_dmg_data["Totally"].sum()` is equal to `merged_total_damage_df["Totally"].sum()`, as the total number of damaged buildings should be conserved.

Reply:
Are you referring to [this](https://drive.google.com/drive/u/2/folders/1ZE2KK0jT7On0GlVeLVi7CLHpjcReTp_O) file? That was created by 510, I would just drop the ID column and then drop any duplicates